http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/KubernetesServiceInterp.scala

```diff
@@ -97,8 +101,11 @@ class LeoKubernetesServiceInterp[F[_]: Parallel](
       else F.unit
 
       clusterId = saveClusterResult.minimalCluster.id
-      saveNodepool <- F.fromEither(getUserNodepool(clusterId, userInfo, req, ctx.now))
-      nodepool <- nodepoolQuery.saveForCluster(saveNodepool).transaction
+      claimedNodepoolOpt <- nodepoolQuery.claimNodepool(clusterId).transaction
+      nodepool <- claimedNodepoolOpt.fold(for {
+        saveNodepool <- F.fromEither(getUserNodepool(clusterId, userInfo, req.kubernetesRuntimeConfig, ctx.now))
+        savedNodepool <- nodepoolQuery.saveForCluster(saveNodepool).transaction
+      } yield savedNodepool)(n => log.info(s"claimed nodepool ${n.id}") >> F.pure(n))
```

Comment:
Might be good to put more information in the log statement, like `"Claimed nodepool {} for {project} / {app}"`

Reply:
Also I had a bit of trouble reading this, maybe a `match` statement would be clearer than `fold`